services-matching/src/test/java/com/mapbox/api/matching/v5/MapboxMapMatchingTest.java

```diff
@@ -224,6 +224,19 @@ public void mapMatchingToDirectionsRoute() throws Exception {
     assertNotNull(mapMatching.executeCall().body().matchings().get(0).toDirectionRoute());
   }
 
+  @Test
+  public void toDirectionsRoute_routeIndex() throws Exception {
+    MapboxMapMatching mapMatching = MapboxMapMatching.builder()
+      .profile(PROFILE_DRIVING)
+      .coordinates(coordinates)
+      .baseUrl(mockUrl.toString())
+      .accessToken(ACCESS_TOKEN)
+      .build();
+    assertEquals(
+      "0",
```

Comment:
Don't you also want to test when it's non zero? I imagine the situation when such a test would have caught a bug: instead of using the for with i you leave forEach, introduce i and forget to increment it:
```
int i = 0;
for (MapMatchingMatching matching : matchings) {
   ...routeIndex(String.valueOf(i))
}
```